open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3818,176 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - table-scan-context
+      properties:
+        table-scan-context:
+          $ref: '#/components/schemas/TableScanContext'
+
+    PlanTableRequest:
+      type: object
+      required:
+        - table-scan-context
```

Comment:
Because the context payload is required in both cases, I don't think it needs to be a sub-field (or should be).

Reply:
Not sure I follow on this, are you saying that we should not have this table scan context field and instead move the properties up to the requests themselves?

Reply:
Yes, exactly. The additional nesting isn't needed.

Reply:
Im ok with also moving this up, would like to confirm with @jackye1995 on his thoughts for this.

Reply:
Will move these fields up.